src/main/java/htsjdk/samtools/cram/build/CramIO.java

```diff
@@ -88,7 +90,7 @@ public static long writeCramEOF(final CRAMVersion cramVersion, final OutputStrea
             throw new RuntimeIOException(e);
         }
 
-        throw new IllegalArgumentException(String.format("Unrecognized CRAM version %s", cramVersion));
+        throw new CRAMException(String.format("Unrecognized CRAM version %s", cramVersion));
```

Comment:
This one is borderline, but I'd err on leaving it as IllegalArgumentException, or else RuntimeException.